src/components/file-uploader/FileUploader.vue

```diff
@@ -0,0 +1,927 @@
+        case CommonImageError.CannotOpen:
+          return 'Cannot open file.';
+        case CommonImageError.PDFnotSupported:
+          return 'This PDF file is not supported.';
+        case CommonImageError.CannotOpenPDF:
+          return 'Cannot open PDF file.'
+        default:
+          return 'An error has occurred.';
+      }
+    },
+
+    /**
+     * Reset input fields so that user can delete a file and
+     * immediately upload that file again.
+     */
+    resetInputFields: function() {
+        this.$refs.browseFileRef.value = '';
+    },
+
+    deleteImage: function(imageModel) {
+        this.resetInputFields();
+        const index = this.images.findIndex(x => x.uuid === imageModel.uuid);
+        this.images.splice(index, 1);
+
+        // If there are no images yet, we have to reset the input so it triggers 'required'.
+        if ( this.required && this.images.length <= 0 ) {
+            console.log('No images, resetting input');
+        }
+        this.$emit('input', this.images);
+    },
+
+    /**
+     * Log image attributes
+     * @param imageModel
+     */
+    logImageInfo: function(title, applicationId, imageModel, additionalInfo) {
+
+        // TODO!
+        // // create log entry
+        // const log: LogEntry = new LogEntry();
+        // log.applicationId = applicationId;
+        // const now = moment();
+        // log.mspTimestamp = now.toISOString();
+        // log.applicationPhase = title + ':  imageModelId: ' + imageModel.id
+        //     + '  imageModelUuid: ' + imageModel.uuid
+        //     + '  imageModelSize: ' + imageModel.size
+        //     + '  imageModelWidth: ' + imageModel.naturalWidth
+        //     + '  imageModelHeight: ' + imageModel.naturalHeight
+        //     + '  imageModelContentType: ' + imageModel.contentType
+        //     + (additionalInfo ? '  ' + additionalInfo : '');
+
+        // // send it while subscribing to response
+        // this.logService.logIt(log, title).subscribe(
+        //     (response) => {
+        //         // console.log('log rest service response: ');
+        //         // console.log(response);
+        //     },
+        //     (error) => {
+        //         console.log('HTTP error response from logging service: ');
+        //         console.log(error);
+        //     },
+        //     () => {
+        //         // console.log('log rest service completed!');
+        //     }
+        // );
+    },
+
+
+
+    /**
+     * Return true if the image size is within range
+     * @param file
+     */
+    checkImageDimensions: function(file) {
+        if (file.naturalHeight < 0 ||
+            file.naturalWidth < 0 ) {
+            return false;
+        }
+        return true;
+    },
+
+    isValid: function() {
+        console.log('isValid', this.images);
+        if (this.required) {
+            return this.images && this.images.length > 0;
+        }
+        return true;
+    }
+  }
+}
+</script>
+
+<style scoped>
```

Comment:
so far in this project we have created separate .css files for the styling, I know vue suggests to put styling in .vue file but wondering if we can have all components consistent